function-session/src/main/java/com/future/function/session/resolver/SessionResolver.java

```diff
@@ -117,6 +117,7 @@ private Session reSetSecurityContextHolderAuthentication(Session session) {
       new UsernamePasswordAuthenticationToken(session.getUserId(), "");
     SecurityContextHolder.getContext()
       .setAuthentication(authentication);
+    SecurityContextHolder.setStrategyName(SecurityContextHolder.MODE_INHERITABLETHREADLOCAL);
```

Comment:
Does `AuthServiceImpl` need to set strategy name too?

Reply:
yes since the completablefuture make a new thread, if the strategy name is not set like this, the authentication from security context holder will not be set in the new thread. basically, this strategy sets the authentication in global mode -> all thread will able to access it.

Reply:
Maybe change it in this PR as well? @oliverrsebastian

Reply:
will see later